sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/SierraItemMergerFeatureTest.scala

```diff
@@ -0,0 +1,109 @@
+package uk.ac.wellcome.platform.sierra_item_merger
+
+import com.twitter.finatra.http.EmbeddedHttpServer
+import com.twitter.inject.server.FeatureTestMixin
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{FunSpec, Matchers}
+import uk.ac.wellcome.models.aws.SQSMessage
+import uk.ac.wellcome.models.{MergedSierraRecord, SierraItemRecord}
+import uk.ac.wellcome.platform.sierra_item_merger.locals.DynamoDBLocal
+import uk.ac.wellcome.platform.sierra_item_merger.utils.SierraItemMergerTestUtil
+import uk.ac.wellcome.test.utils.{AmazonCloudWatchFlag, SQSLocal}
+import uk.ac.wellcome.utils.JsonUtil
+import uk.ac.wellcome.dynamo._
+
+class SierraItemMergerFeatureTest
```

Comment:
The tests as written are fine, but I think we need more of them. At a minimum, corresponding tests to the ones we have on the BibMergerFeatureTest:

* should update an item in DynamoDB if a newer version is sent to SQS
* should not update an item in DynamoDB if an older version is sent to SQS
* should put an item from SQS into DynamoDB if the ID exists but no itemData

I wonder, as well – do we want a test around interactions between bib/item data?

i.e. the item merger should not change bibData when it updates the items, and vice versa for bibs

Reply:
I don't think there is much value in replicating test at the feature level. Those cases, unless I misunderstand their scope, are covered by the `SierraItemMergerUpdaterServiceTests`.

Feature tests should not cover corner cases but just assert that thing are wired in correclty and that everything works together.

Reply:
Got It – I’d missed the service tests when I did my original review.

Reply:
I'm inclined to agree that we don't need to cover these at multiple levels.